### ISSUE TITLE
Opening block from search result does not jump to the block

### ISSUE THREAD

**Issue by agentydragon** [athensresearch/athens#1063]
**Problem**

Say you have a large page, several viewports long. Say in the middle of it you have a block saying "cheese". Say you go to Daily Notes, do `Ctrl+K`, type in `cheese`, the block shows up in the search result. You press Enter.

Then, Athens opens the correct page. But it does not scroll to the correct block, or highlight it.

Now while you are on the right page, you still have to manually search for the block you're looking for within the page.

**Athens Version**

v1.0.0-beta.77-5-g454682547

**Comment by agentydragon** [comment]
There's an analogous problem when you zoom in into a single block, and then zoom out. After the zoom-out, the block you were in previously is not highlighted in any way.

**Comment by shanberg** [comment]
I recommend that any solution to this consider scrolling to targeted blocks using an anchor link (something like `url/#blockid`), styling the targeted block using the `::target` CSS selector to provide the highlight effect, and adding `scroll-behavior: smooth;` to the scrolling container to smooth the scrolling event.

**Comment by agentydragon** [comment]
> I recommend that any solution to this consider scrolling to targeted blocks using an anchor link (something like `url/#blockid`), styling the targeted block using the `::target` CSS selector to provide the highlight effect, and adding `scroll-behavior: smooth;` to the scrolling container to smooth the scrolling event.

Personally I'd prefer to not have the scroll smoothed, because I find animations usually just slow me down / distract me, but if we get good customizability, I might just flip a switch to disable it :) (there's a discussion somewhere)

Otherwise that's exactly how Id implement this :)

**Comment by shanberg** [comment]
> might just flip a switch to disable it :) (there's a discussion somewhere)
> 
> Otherwise that's exactly how Id implement this :)

@agentydragon You're right, be sure to account for `prefers-reduced-motion` and perhaps a toggle in app settings.

Elaborating for completeness's sake: I'd keep the animation on by default if we're simply scrolling on the page, because it helps orient users to where they've gone relative to where they were, and avoids disorenting jumps. Highly opinionated users (I know there are a lot of you) should know to look in settings to adjust this.

If we're jumping to a block on a new page, I don't think any scrolling transition is necessary, because the jump in context is not purely a jump in ~vertical location.

**Comment by agentydragon** [comment]
As an alternative to scrolling/extra usability hint, we could also have
highlights for "this is where you just jumped to, this is where you were
before", maybe as if you had Vim's jump buffer and highlighted the last K
places in it.

**Comment by shanberg** [comment]
> As an alternative to scrolling/extra usability hint, we could also have highlights for "this is where you just jumped to, this is where you were before", maybe as if you had Vim's jump buffer and highlighted the last K places in it.

by "highlight places in the jump buffer" do you mean mark the scrollbar with ranges indicating past jump points?

**Comment by agentydragon** [comment]
I meant highlighting the blocks themselves - for example by a different
background color. E.g., lighter blue for further down the "jump buffer",
darker blue for more recent jumps.

**Comment by shanberg** [comment]
> I meant highlighting the blocks themselves - for example by a different background color. E.g., lighter blue for further down the "jump buffer", darker blue for more recent jumps.

@agentydragon That sounds really interesting. Is there a video or screenshot you could provide to illustrate?

**Comment by juniusfree** [comment]
@shanberg @agentydragon how about opening it as a block page?

**Comment by juniusfree** [comment]
Here's an example of jump to highlight. Works for Chrome only: https://en.wikipedia.org/wiki/Potoooooooo#:~:text=He%20is%20now%20best%20known%20for%20the%20unusual%20spelling%20of%20his%20name%2C%20pronounced%20Potatoes
